saml-metadata-bindings/src/main/java/uk/gov/ida/saml/metadata/EidasMetadataResolverRepository.java

```diff
@@ -63,16 +65,19 @@ public MetadataResolver getMetadataResolver(String entityId) {
         return metadataResolvers;
     }
 
+    public List<String> getTrustAnchorsEntityIds() {
+        return trustAnchors.stream().map(JWK::getKeyID).collect(Collectors.toList());
+    }
+
     private void refresh() {
         delayBeforeNextRefresh =  eidasMetadataConfiguration.getTrustAnchorMaxRefreshDelay();
         try {
-            List<JWK> trustAnchors = trustAnchorResolver.getTrustAnchors();
-
-            removeMetadataResolvers(trustAnchors);
+            trustAnchors = trustAnchorResolver.getTrustAnchors();
+            removeMetadataResolvers();
             registerMetadataResolvers(trustAnchors);
         } catch (Exception e) {
             log.error("Error fetching trust anchor or validating it", e);
-            delayBeforeNextRefresh = eidasMetadataConfiguration.getTrustAnchorMinRefreshDelay();
+            setShortRefreshDelay();
```

Comment:
Why do we need to call this?

Reply:
So we were just following the metadata resolver's way of doing things. There's a short delay and a long one. If everything's fine, use the long delay. If there was a problem or something's expiring soon, use the short one.

This commit's just putting it into a private method.

Happy to make any changes you suggest.